Automattic/wp-calypso
# Hide the Free plan if a coupon is provided as a dependency

Related to #

## Proposed Changes

* In https://github.com/Automattic/wp-calypso/pull/85551, we introduced the ability to add a coupon code as a query dependency, so that the coupon can be auto-applied at checkout. If a coupon is supplied, then it shouldn't be required to show a free plan. This PR hides the free plan if a coupon exists as a signup dependency.
* It's assumed here that the coupon will apply to paid plans at the minimum and I think that's a reasonable assumption to make? 

## Testing Instructions



* Go to `/start/?coupon=NEWYEAR2024` and verify that the Free plan is hidden in the plans step. Also verify that account creation and site creation works correctly.
* Go to `/start` and verify that account creation and site creation works correctly.
